test/Array/pop5.js

```diff
@@ -0,0 +1,34 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Copyright (c) 2022 ChakraCore Project Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+const array = [1, 2, 3];
+
+const testRuns = 10;
+let counter = 0;
+
+let proto = [];
+Object.defineProperty(proto, 3, {
+    get: function () {
+        counter++;
+        return 42;
+    }
+});
+array.__proto__ = proto;
+
+function hotFunction() {
+    // Store return value
+    const x = array.pop();

```

Comment:
The jit could theoretically optimise away this store (detect that x is a local that's never read and therefore skip the assignment), please could you store in a global variable instead to prevent that?

Reply:
I was concerned about that too, but I was not sure, how to handle this…